code/src/layouts/cart-drawer/CartDrawer.tsx

```diff
@@ -9,17 +9,47 @@ import AppIconButton from "@/components/app-icon-button/AppIconButton";
 import AppTypography from "@/components/app-typography/AppTypography";
 
 import { useDrawerContext } from "@/context/DrawerContext";
+import useSnackbar from "@/hooks/use-snackbar/useSnackbar";
+import { useRemoveFromCartMutation } from "@/store/api/cartApi";
+import { useUserDetailsSelector } from "@/store/slices/userSlice";
+import { CartItem } from "@/types/cart.types";
 import formatPrice from "@/utils/format-price/formatPrice";
 
 import "@/layouts/cart-drawer/CartDrawer.scss";
 
 const CartDrawer = () => {
-  const cartItems = mockCartItems.map((item) => (
-    <CartDrawerItem key={item.productId} {...item} />
-  ));
+  const user = useUserDetailsSelector();
+
+  const [removeItem] = useRemoveFromCartMutation();
```

Comment:
Same loading state here

Reply:
I am not sure how to display loading indicator for the user in this case. Maybe you can suggest something?

![image](https://github.com/idx-academy/spa-orders/assets/62070431/2a547173-587e-491f-8b37-35e1bce86af4)

Reply:
Yeah, maybe we can leave it for now and discuss if we need this feature